src/server/protocol.ts

```diff
@@ -442,6 +442,12 @@ namespace ts.server.protocol {
           * Errorcodes we want to get the fixes for.
           */
         errorCodes?: number[];
+
+        /**
+         * An optional set of settings to be used when computing indentation.
+         * If argument is omitted - then it will use settings for file that were previously set via 'configure' request or global settings.
+         */
+        options?: EditorSettings;

```

Comment:
@mjbvz I didn't integrate all the relevant formatting options, focusing just on indentation, but looking now we ought to follow directives for `FormattingSettings` like `insertSpaceAfterCommaDelimiter` or `insertSpaceBeforeFunctionParenthesis`. I will look into whether we ought to (i) have an adhoc solution of hand-writing the formatting into each quickfix or (ii) extend the formatter API on the script side to also work with "speculative insertions" rather than just inspecting a range of the current file.